src/tls_stream.rs

```diff
@@ -352,6 +352,20 @@ impl<S> TlsStream<S>
         self.context.remote_cert()
     }
 
+    /// Returns whether or not the session was resumed.
+    ///
+    /// # Remarks
+    /// This is only valid if the TLS handshake is completed, and will return None if called
+    /// before this point.
+    pub fn session_resumed(&self) -> io::Result<Option<bool>> {
+        if let State::Streaming { .. } = self.state {
```

Comment:
Is there a reason for this check? 
E.g. would calling `self.context.session_info()` similarily to `remote_cert` fail, if the handshake wasn't completed yet?

Reply:
How is `remote_cert()` expected to fail if the handshake wasn't completed?

It looks like `self.context.session_info()` will fail with `SEC_E_INVALID_HANDLE` before the handshake completes.

I'm okay with changing this to just `pub fn session_resumed(&self) -> io::Result<bool>` if the above matches existing expected behavior.

Reply:
It looks like `remote_cert()` fails with the same error. I'll remove the check.